### DISCUSSION TITLE
Seeking Guidance: Developing Cross-Platform Mobile App with Voice Recognition

### DISCUSSION THREAD

**Discussion by LaoSun2023** [community/community#65967] [Programming Help]
### Body

hey there,I am a beginner and I need to develop a mobile app that is compatible with both iOS and Android operating systems. The app should have a front-end UI development based on lock screen and screensaver (widget), control over the device's microphone and speaker, and integration with an external voice recognition engine with performance optimization. Can you recommend a development platform and programming language that is easy to learn and offers good runtime efficiency?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by mehulrajdev** [comment]
Koltlin is the best one and you can use Android studio

https://kotlinlang.org/

https://developer.android.com/studio?gclid=Cj0KCQjwgNanBhDUARIsAAeIcAv18r-bbL_gtQLNlSI49gp_xXO0yDn-wMSq-PBj50n_m_LNB88m_wIaArFEEALw_wcB&gclsrc=aw.ds
